### ISSUE TITLE
Fails to build when DEV_MODE is set to False

### ISSUE THREAD

**Issue by shtayeb** [protibimbok/django-vite-plugin#18]
Hi, thank you for the module.

I have been using this plugins in development without facing any issues but I have encountered problems with the build process when attempting to host it on [Railway](https://railway.app/).

## Steps to reproduce.

Plugin settings

```python
DJANGO_VITE_PLUGIN = {
    "BUILD_DIR": "staticfiles/build",
    "BUILD_URL_PREFIX": "/" + STATIC_URL + "build",
    "DEV_MODE": False,
}
```
## The problem
During the execution of `npm run build`, if DEV_MODE is set to False, the process searches for a manifest.json file which is not yet created in the production environment. This results in a build failure and the bellow error is shown.

```shell
File "/opt/venv/lib/python3.8/site-packages/django_vite_plugin/utils.py", line 33, in <module>
﻿#17 1.851     raise RuntimeError(
﻿#17 1.851 RuntimeError: Cannot read Vite manifest file at /app/staticfiles/build/manifest.json : [Errno 2] No such file or directory: '/app/staticfiles/build/manifest.json'
```

If I keep DEV_MODE=True assets will be loaded using vite server which doesnt exist in the production

## The repository

Here is the [repo link ](https://github.com/shtayeb/Django-Studybudy-YT)

The same problem also occurs in the example/demo project as well.

I am new to Django so forgive me if it has an obvious answer. 

Thanks in advance

**Comment by shtayeb** [comment]
Hi Again !

I found a workaround for my issue.

## My workaround:

I have two settings
- local.py
  In this settings file the DEV_MODE is True
- production.py
  In this settings file the DEV_MODE is False. This settings is used in the production server.

In the `vite.config.js`  I used the local.py settings to build the assets, since it builds without any error and serving the assets is done with DEV_MODE to False.

```javascript
import { defineConfig } from "vite";
import djangoVite from "django-vite-plugin";

export default defineConfig({
  plugins: [
    djangoVite({
      input: ["./static/js/main.js", "./static/styles/tailwind.css"],
      // settings file with DEV_MODE=True
      pyArgs: ["--settings", "core.settings.local"],
    }),
  ],
});

```

**Comment by protibimbok** [comment]
Thank you for the report. It's a nice workaround.
And alhamdulillah, it's fixed now.
Please update the `django-vite-plugin` python package.
```bash
pip install --upgrade django-vite-plugin
```

Happy Coding

**Comment by hfroot** [comment]
I have the same problem. Using `django_vite_plugin` version 3.0.0 (python) and `django-vite-plugin` 4.0.1 (js).
The host we use allows `python manage.py` commands to be run, and we can define them in our app, but I come up against the "Cannot read Vite manifest file" error presumably because the app is loaded with `django_vite_plugin` in the list of `INSTALLED_APPS`. 
Anyway, I can run `npm run build` locally with `DEV_MODE` as `False` so I am not sure if this problem is in the scope of this project. I am looking into solutions on the side of our hosting service.

**Comment by hfroot** [comment]
Looking again this morning, I do find that when trying to build for the first time with `DEV_MODE` as `False` and `npm run build` I get the error 

```
  File "/.../venv/lib/python3.10/site-packages/django_vite_plugin/management/commands/django_vite_plugin.py", line 7, in <module>
    from ...utils import find_asset
  File "/.../venv/lib/python3.10/site-packages/django_vite_plugin/utils.py", line 30, in <module>
    raise RuntimeError(
RuntimeError: The Vite manifest file does not exist at: build/.vite/manifest.json
```

I can see that the change introduced by #19 is no longer there. Is it possible to reinstate it?

**Comment by protibimbok** [comment]
Could you please check and confirm if #59 fixes the problem?

**Comment by hfroot** [comment]
Hi @protibimbok thanks for the fix.

I find that it works now when I am building using the npm command directly. 

However, when I try and use the custom python command, it throws the same error about the vite manifest file. This is a non-blocking issue because we have a workaround.

By custom command, I mean that in the same Django directory as our `settings.py` file, we have added a file `management/commands/buildnpm.py` which looks like

```
import os
import subprocess
from django.core.management.base import BaseCommand

BASE_DIR = os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))))


class Command(BaseCommand):
    help = "Fetches npm dependencies and makes a prod build of the frontend application"

    def handle(self, *args, **options):
        print(BASE_DIR)

        os.chdir(os.path.join(BASE_DIR, "2024-frontend"))
        subprocess.run(["npm", "install"])
        subprocess.run(["npm", "run", "build"])
        os.chdir(os.path.join(BASE_DIR))
```

This allows us to run these commands on our production server, which does not support running npm commands directly.

Our workaround is:
- create a bash script that copies a dummy manifest file into `build/.vite` and `2024-frontend/build/.vite`
- get our server to run this before a deployment
- then the server runs the python commands `python manage.py buildnpm` as well as the others we need

And that works fine too.

**Comment by protibimbok** [comment]
Do you mean the error is happening with `DEV_MODE = True`?

**Comment by hfroot** [comment]
No, with DEV_MODE = False

**Comment by protibimbok** [comment]
Alhamdulillah, this one is fixed in v3.0.4
